client/client_test.go

```diff
@@ -1876,6 +1876,14 @@ func TestClient_hasLocalState(t *testing.T) {
 
 	c.stateDB = cstate.NewMemDB(c.logger)
 
+	t.Run("nil Job", func(t *testing.T) {
+		alloc := mock.BatchAlloc()
+		alloc.Job = nil
+		c.stateDB.PutAllocation(alloc)
+
+		require.False(t, c.hasLocalState(alloc))
```

Comment:
We're slowly migrating all our uses of `testify` to [`shoenig/test`](https://pkg.go.dev/github.com/shoenig/test) instead, which fixes some correctness issues around equality testing. So let's make sure we're doing that on new tests:

```suggestion
		must.False(t, c.hasLocalState(alloc))
```